osu.Game.Tests/Chat/TestSceneChannelManager.cs

```diff
@@ -83,12 +89,62 @@ public void TestCommandsPostedToCorrectChannelWhenNotCurrent()
             AddAssert("/np command received by channel 2", () => channel2.Messages.Last().Content.Contains("is listening to"));
         }
 
+        [Test]
+        public void TestMarkAsReadIgnoringLocalMessages()
+        {
+            Channel channel = null;
+
+            AddStep("join channel and select it", () =>
+            {
+                channelManager.JoinChannel(channel = createChannel(1, ChannelType.Public));
+                channelManager.CurrentChannel.Value = channel;
+            });
+
+            AddStep("post message", () => channelManager.PostMessage("Something interesting"));
+            AddUntilStep("wait until the message is posted", () => channel.Messages.Count == sentMessages.Count);
+            AddStep("post /help command", () => channelManager.PostCommand("help", channel));
+            AddStep("post /me command with no action", () => channelManager.PostCommand("me", channel));
+            AddStep("post /join command with no channel", () => channelManager.PostCommand("join", channel));
+            AddStep("post /join command with non-existent channel", () => channelManager.PostCommand("join i-dont-exist", channel));
+            AddStep("post non-existent command", () => channelManager.PostCommand("non-existent-cmd arg", channel));
+
+            AddStep("register mark channel as read request handler", () =>
+            {
+                ((DummyAPIAccess)API).HandleRequest = req =>
+                {
+                    switch (req)
+                    {
+                        case MarkChannelAsReadRequest markRead:
+                            var isSentMessage = sentMessages.Contains(markRead.Message);
+
+                            AddAssert("mark channel as read called with a real message", () => isSentMessage);
```

Comment:
Assert steps can't be added like this from anywhere, it will execute after every other test step and not inline here. And it was expected that it should be added last, then it should be in the outer scope, and the flag should be in a field (and reset on test setup).